frontend/javascripts/viewer/view/action-bar/user_scripts_modal_view.tsx

```diff
@@ -15,140 +15,113 @@ type UserScriptsModalViewProps = {
   onOK: (...args: Array<any>) => any;
   isOpen: boolean;
 };
-type State = {
-  code: string;
-  isCodeChanged: boolean;
-  scripts: Array<Script>;
-  selectedScript: string | null | undefined;
-  isLoading: boolean;
-};
-
-class _UserScriptsModalView extends React.PureComponent<UserScriptsModalViewProps, State> {
-  state: State = {
-    code: "",
-    isCodeChanged: false,
-    scripts: [],
-    // Needs to be undefined so the placeholder is displayed in the beginning
-    selectedScript: undefined,
-    isLoading: true,
-  };
 
-  componentDidMount() {
-    this.fetchData();
-  }
+const _UserScriptsModalView: React.FC<UserScriptsModalViewProps> = ({ onOK, isOpen }) => {
+  const [code, setCode] = React.useState("");
+  const [isCodeChanged, setIsCodeChanged] = React.useState(false);
+  const [scripts, setScripts] = React.useState<Script[]>([]);
+  const [selectedScript, setSelectedScript] = React.useState<string | null>(null);
+  const [isLoading, setIsLoading] = React.useState(true);
 
-  fetchData() {
+  React.useEffect(() => {
```

Comment:
Maybe `useGuardedFetch` would be better here to already have an `isLoading` prop included. It also includes catching errors.